inc/ctest.h

```diff
@@ -204,55 +204,68 @@ typedef void* void_ptr;
-extern C_LINKAGE int int16_t_Compare(int16_t left, int16_t right);
-extern C_LINKAGE int uint32_t_Compare(uint32_t left, uint32_t right);
-extern C_LINKAGE int int32_t_Compare(int32_t left, int32_t right);
-extern C_LINKAGE int uint64_t_Compare(uint64_t left, uint64_t right);
-extern C_LINKAGE int int64_t_Compare(int64_t left, int64_t right);
-#endif
 
 extern C_LINKAGE char* ctest_sprintf_char(const char* format, ...);
 extern C_LINKAGE void ctest_sprintf_free(char* string);
 
+#define CTEST_DECLARE_EQUALITY_ASSERTION_FUNCTIONS_FOR_TYPE(type) \
+extern C_LINKAGE void MU_C2(type,_AssertAreEqual)(type left, type right, char* ctest_message); \
+extern C_LINKAGE void MU_C2(type,_AssertAreNotEqual)(type left, type right, char* ctest_message);
+
+#define CTEST_ASSERT_ARE_EQUAL_IMPL_FOR_TYPE(type) \
+    char expectedString[1024]; \
+    char actualString[1024]; \
+    MU_C2(type,_ToString)(expectedString, sizeof(expectedString), left); \
+    MU_C2(type,_ToString)(actualString, sizeof(actualString), right); \
+    if (MU_C2(type,_Compare)(left, right)) \
+    { \
+        LogError("  Assert failed in line %d %s Expected: %s, Actual: %s\n", __LINE__, (ctest_message == NULL) ? "" : ctest_message, expectedString, actualString); \
+        ctest_sprintf_free(ctest_message); \
+        if (g_CurrentTestFunction != NULL) *g_CurrentTestFunction->TestResult = TEST_FAILED; \
+        do_jump(&g_ExceptionJump, expectedString, actualString); \
+    } \
+    ctest_sprintf_free(ctest_message);
+
+#define CTEST_ASSERT_ARE_NOT_EQUAL_IMPL_FOR_TYPE(type) \
+    char expectedString[1024]; \
+    char actualString[1024]; \
+    MU_C2(type,_ToString)(expectedString, sizeof(expectedString), left); \
+    MU_C2(type,_ToString)(actualString, sizeof(actualString), right); \
+    if (!MU_C2(type,_Compare)(left, right)) \
+    { \
+        LogError("  Assert failed in line %d %s Expected: %s, Actual: %s\n", __LINE__, (ctest_message == NULL) ? "" : ctest_message, expectedString, actualString); \
+        ctest_sprintf_free(ctest_message); \
+        if (g_CurrentTestFunction != NULL) *g_CurrentTestFunction->TestResult = TEST_FAILED; \
+        do_jump(&g_ExceptionJump, expectedString, actualString); \
+    } \
+    ctest_sprintf_free(ctest_message);
+
+#define CTEST_DEFINE_STATIC_EQUALITY_ASSERTION_FUNCTIONS_FOR_TYPE(type) \
+static void MU_C2(type,_AssertAreEqual)(type left, type right, char* ctest_message) \
+{ \
+    CTEST_ASSERT_ARE_EQUAL_IMPL_FOR_TYPE(type) \
+} \
+static void MU_C2(type,_AssertAreNotEqual)(type left, type right, char* ctest_message) \
+{ \
+    CTEST_ASSERT_ARE_NOT_EQUAL_IMPL_FOR_TYPE(type) \
+}
+
+#define CTEST_DEFINE_EXTERN_EQUALITY_ASSERTION_FUNCTIONS_FOR_TYPE(type) \
+void MU_C2(type,_AssertAreEqual)(type left, type right, char* ctest_message) \
+{ \
+    CTEST_ASSERT_ARE_EQUAL_IMPL_FOR_TYPE(type) \
+} \
+void MU_C2(type,_AssertAreNotEqual)(type left, type right, char* ctest_message) \
+{ \
+    CTEST_ASSERT_ARE_NOT_EQUAL_IMPL_FOR_TYPE(type) \
+}
```

Comment:
This can be extracted in a CTEST_DEFINE_EQUALITY_ASSERTION_FUNCTIONS_COMMON macro which can have an argument that is either static or empty token, so that the code duplication would not exist.